keps/sig-cluster-lifecycle/kubeadm/2568-kubeadm-non-root-control-plane/README.md

```diff
@@ -0,0 +1,1022 @@
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a security conscious user I would like to run the kubernetes control-plane as non-root.
+
+#### Story 2
+
+As a user of kubeadm which sells it self as the best-practice and fast way to bootstrap the cluster, I expect it to follow security best practices like running containers as non-root.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+### How do we assign these uid/gids to the components?
+There are 2 options for setting the uid/gid of the control-plane components:-
+
+1. **Update the kubeadm API to make the uid/gid configurable by the user:**  This can be implemented in 2 ways:-
+    1. This can be done by adding fields `UID` and `GID` of type `int64` to the `ControlPlaneComponent` `struct` in https://github.com/kubernetes/kubernetes/blob/854c2cc79f11cfb46499454e7717a86d3214e6b0/cmd/kubeadm/app/apis/kubeadm/types.go#L132 as demonstrated in [this](https://github.com/kubernetes/kubernetes/pull/99753) PR.
+    2. User provides kubeadm with a range of uid/gids that are safe to use i.e. are not being used by another user.
+
+2. **Use constant values for uid/gid:** The `UID` and `GID`for each of the control-plane components would be set to some predetermined value in https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/constants/constants.go.
+
+The author(s) believes that allowing the user to set the UID and GID throught the API is more user friendly as there my be some constraints on what UID/GID can be used on the users machine. The reponsibility of making sure that the uid/gid is not in use or the range is reserved only for kubeadm is on the user. This keeps the desing of the kubeadm simple and makes the contract explicit.
```

Comment:
What about having sane but configurable defaults? We should remove the friction of running these components as non-root by default, but if users want to configure UID and GID for each component that should be fine as well.